### PR TITLE
fix(VBtn): allow `type` and `disabled` for non-button buttons

### PR DESCRIPTION
**Pull request by dougalg**
Sometimes we want button styles on non-button elements, such as an input, or other. Some of these elements support the `type` and/or `disabled` attributes, but currently vuetify does not allow them to get rendered unless VBtn is a "button".

https://github.com/vuetifyjs/vuetify/issues/11289

## Description

Resolve https://github.com/vuetifyjs/vuetify/issues/11289

There's no reason to explicitly remove `type` and `disabled` for non-button elements. If the concern is extraneous tags for cases where the attributes don't work, then it may make sense to list all tags that support `disabled` and all tags that support `type`, although this feels not very scalable.

## Motivation and Context

This allows the use of the semantic `<input type="submit"..` element which visually is rendered as a button.

## How Has This Been Tested?

unit tests

## Markup:

<details>

```vue
<div id="app">
  <v-app>
    <v-btn
      tag="input"
      type="submit"
      value="Log In"
      name="login"
      elevation="0"
    />
  </v-app>
</div>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by nekosaur** [comment]
not sure I see the harm in just adding the attributes regardless of what type of element.

**Comment by dougalg** [comment]
I agree, would prefer to just allow all types. I think the original bug was that because `type` is set by default to `button`, it was showing up as `type="button"` in places where it shouldn't, which is an accessibility issue. 

Most likely, that could be dealt with in a simpler way. We always set the type, but we only default the `type` to `button` if the component is a button, otherwise the implementer must pass it explicitly.

@johnleider what do you think? That way I think I could get rid of those constants at the top of the file.

File: packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts [hunk 1]
```diff
@@ -96,6 +96,17 @@ describe('VBtn.ts', () => {
     expect(wrapper.html()).toMatchSnapshot()
   })
 
+  it('should should render disabled on an <a> tag when using href prop', () => {
```
Comment (nekosaur) [inline packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts:99]: double should

File: packages/vuetify/.eslintrc.js [hunk 2]
```diff
@@ -30,10 +30,18 @@ module.exports = {
     'jest/prefer-to-contain': 'warn',
     'jest/prefer-to-have-length': 'warn'
   },
-  overrides: [{
-    files: 'dev/Playground.vue',
-    rules: {
-      'max-len': 'off'
+  overrides: [
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/.eslintrc.js:33]: that should be rather a separate PR, also i'm not sure if this is good idea, if there's too many `it(...)` in one `describe()` maybe it's a sign that they should be split into separate `discrabe()`s

Comment (dougalg) [reply packages/vuetify/.eslintrc.js:33]: That's fair. I'll revert this. Any suggestions on how to split up the describes, or do you know of any other test files that I could look at to get an idea?

File: packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts (hunk 1 above)
Comment (dougalg) [reply packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts:99]: Because I believe it so strongly! Hahah, thanks, will fix.

File: packages/vuetify/.eslintrc.js (hunk 2 above)
Comment (jacekkarczmarczyk) [reply packages/vuetify/.eslintrc.js:33]: VIcon.spec.ts, colorUtils.spec.ts etc

**Comment by dougalg** [comment]
Closing this, as it seems, based on discussion in the initial bug report that it isn't really useful.
